File: articles/moral-psychology-in-organizations-and-institutions/cpp/institutional_integrity_score.cpp
/*
Synthetic institutional integrity score example.
Educational scaffold only.

Compile:
  c++ -std=c++17 cpp/institutional_integrity_score.cpp -o outputs/tables/institutional_integrity_score_cpp
Run:
  ./outputs/tables/institutional_integrity_score_cpp
*/

#include <iostream>

int main() {
    double accountability_strength = 0.8;
    double speak_up_protection = 0.7;
    double value_alignment = 0.9;
    double cultural_climate = 0.6;
    double corruption_pressure = 0.3;
    double diffusion_responsibility = 0.4;

    double integrity =
        0.30 * accountability_strength +
        0.30 * speak_up_protection +
        0.30 * value_alignment +
        0.20 * cultural_climate -
        0.30 * corruption_pressure -
        0.20 * diffusion_responsibility;

    std::cout << "Synthetic institutional integrity score: "
              << integrity << std::endl;

    return 0;
}

File: articles/moral-psychology-in-organizations-and-institutions/rust/institutional_integrity_score.rs
// Synthetic institutional integrity score example.
// Educational scaffold only.
//
// Compile:
//   rustc rust/institutional_integrity_score.rs -o outputs/tables/institutional_integrity_score_rust
// Run:
//   ./outputs/tables/institutional_integrity_score_rust

fn main() {
    let accountability_strength = 0.8_f64;
    let speak_up_protection = 0.7_f64;
    let value_alignment = 0.9_f64;
    let cultural_climate = 0.6_f64;
    let corruption_pressure = 0.3_f64;
    let diffusion_responsibility = 0.4_f64;

    let integrity =
        0.30 * accountability_strength
        + 0.30 * speak_up_protection
        + 0.30 * value_alignment
        + 0.20 * cultural_climate
        - 0.30 * corruption_pressure
        - 0.20 * diffusion_responsibility;

    println!("Synthetic institutional integrity score: {:.3}", integrity);
}
